x-pack/plugins/security_solution/public/timelines/store/timeline/epic.ts

```diff
@@ -182,21 +132,20 @@ export const createTimelineEpic =
             myEpicTimelineId.setTemplateTimelineVersion(addNewTimeline.templateTimelineVersion);
             return getOr(false, 'payload.savedTimeline', action);
           } else if (
-            timelineActionsType.includes(action.type) &&
-            !timelineObj.isLoading &&
+            action.type === saveTimeline.type &&
+            // don't save when the timeline (or its related queries are loading), unless the timeline changed
+            (!timelineObj.isLoading || timelineObj.changed) &&
```

Comment:
Yea, this might be the scenario that you run into where you are unable to save the timeline while data is loading. We actually should change this to `isSaving` because there's no need to wait for data to load. It would fix an existing bug, but it's hard to reproduce unless your in an environment under load. We can do it in a separate PR though if that makes you more comfortable

Reply:
Yep this caused all sorts of issues: https://github.com/elastic/kibana/pull/169239/commits/82f741a78a12b382c93854c854ee8fd4a6e9340b